mav_msgs/include/mav_msgs/common.h

```diff
@@ -142,6 +142,27 @@ inline void getEulerAnglesFromQuaternion(const Eigen::Quaternion<double>& q,
   }
 }
 
+// Calculates angular velocity (omega) from rotation vector derivative
+inline Eigen::Vector3d omegaFromRotationVector(
+                            const Eigen::Vector3d& rot_vec,
+                            const Eigen::Vector3d& rot_vec_vel) {
+  double phi = rot_vec.norm();
+  if (abs(phi) < 1.0e-3) {
+    // This captures the case of zero rotation
+    return rot_vec_vel;
+  }
+  // Create skew-symmetric matrix from rotation vector
+  Eigen::Matrix3d phi_matrix;
+  phi_matrix << 0.0f, -rot_vec(2), rot_vec(1),
+                rot_vec(2), 0.0f, -rot_vec(0),
+                -rot_vec(1), rot_vec(0), 0.0f;
+  // Set up matrix to calculate omega
+  Eigen::Matrix3d W;
+  W = Eigen::MatrixXd::Identity(3,3) - phi_matrix * (1-cos(phi))/phi/phi
+      + phi_matrix*phi_matrix * (phi-sin(phi))/phi/phi/phi;
```

Comment:
Typically devision is slower than multiplication. Consider creating `double phi_inv = 1.0 / rot_vec.norm();` and use multiplication instead.